Runtime/Physics/BuoyantObject.cs

```diff
@@ -121,37 +121,37 @@ private void SetupColliders()
 
         private void Update()
         {
+            switch (_buoyancyType)
+            {
+                case BuoyancyType.Physical:
+                case BuoyancyType.PhysicalVoxel:
+                    return;
+            }
+
 #if STATIC_EVERYTHING
             var dt = 0.0f;
 #else
             var dt = Time.deltaTime;
 #endif
+
             switch (_buoyancyType)
             {
                 case BuoyancyType.NonPhysical:
-                {
-                    _samplePoints[0] = transform.position;
-                    var t = transform;
-                    var vec  = t.position;
-                    vec.y = Heights[0].y + waterLevelOffset;
-                    t.position = vec;
-                    var up = t.up;
-                    t.up = Vector3.Slerp(up, _normals[0], dt);
-                    break;
-                }
+                    {
+                        _samplePoints[0] = transform.position;
+                        var t = transform;
+                        var vec  = t.position;
+                        vec.y = Heights[0].y + waterLevelOffset;
+                        t.position = vec;
+                        var up = t.up;
```

Comment:
I know you didn't write the original, but the assignment to up isn't needed. t.up will be evaluated first in the next line before overwriting it. t.up = Vector3.Slerp(**t.up**, _normals[0], dt)